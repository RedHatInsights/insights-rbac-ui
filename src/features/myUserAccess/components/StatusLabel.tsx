import React from 'react';
import { useIntl } from 'react-intl';
import messages from '../../../Messages';

import { Label, Tooltip, TooltipPosition } from '@patternfly/react-core';

interface StatusLabelProps {
  isOrgAdmin?: boolean;
  isUserAccessAdmin?: boolean;
}

const StatusLabel: React.FC<StatusLabelProps> = ({ isOrgAdmin, isUserAccessAdmin }) => {
  const intl = useIntl();

  const tootltipLabel = intl.formatMessage(messages[isOrgAdmin ? 'orgAdministrator' : 'userAccessAdmin']);
  const tooltipContent = <span>{intl.formatMessage(messages[isOrgAdmin ? 'orgAdminHint' : 'userAccessAdminHint'])}</span>;

  if (isOrgAdmin || isUserAccessAdmin) {
    return (
      <Tooltip position={TooltipPosition.right} content={tooltipContent}>
        <Label color="purple"> {tootltipLabel} </Label>
      </Tooltip>
    );
  }
  return <React.Fragment />;
};

export default StatusLabel;
