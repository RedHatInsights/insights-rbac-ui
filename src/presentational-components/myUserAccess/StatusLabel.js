import React from 'react';
import PropTypes from 'prop-types';

import { Label, Tooltip, TooltipPosition } from '@patternfly/react-core';

const StatusLabel = ({ isOrgAdmin }) => (
  <React.Fragment>
    {isOrgAdmin ? (
      <Tooltip position={TooltipPosition.right} content={<span> You can manage other users&apos; permissions with &apos;User access&apos; </span>}>
        <Label color="purple"> Org. Administrator </Label>
      </Tooltip>
    ) : (
      <Label color="purple"> User </Label>
    )}
  </React.Fragment>
);

StatusLabel.propTypes = {
  isOrgAdmin: PropTypes.bool,
};

export default StatusLabel;
