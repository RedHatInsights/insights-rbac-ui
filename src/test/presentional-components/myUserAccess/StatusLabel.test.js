import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import StatusLabel from '../../../presentational-components/myUserAccess/StatusLabel';

describe('<StatusLabel />', () => {
  it('should render correctly', () => {
    expect(toJson(mount(<StatusLabel />))).toMatchSnapshot();
  });

  it('should render correctly as admin', () => {
    expect(toJson(mount(<StatusLabel isOrgAdmin />))).toMatchSnapshot();
  });
});
