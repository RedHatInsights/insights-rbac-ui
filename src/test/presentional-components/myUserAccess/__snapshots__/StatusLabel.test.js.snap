// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusLabel /> should render correctly 1`] = `
<StatusLabel>
  <Label
    color="purple"
  >
    <span
      className="pf-c-label pf-m-purple"
    >
      <span
        className="pf-c-label__content"
      >
         User 
      </span>
    </span>
  </Label>
</StatusLabel>
`;

exports[`<StatusLabel /> should render correctly as admin 1`] = `
<StatusLabel
  isOrgAdmin={true}
>
  <Tooltip
    content={
      <span>
         You can manage other users' permissions with 'User access' 
      </span>
    }
    position="right"
  >
    <Popper
      appendTo={[Function]}
      distance={15}
      enableFlip={true}
      flipBehavior={
        Array [
          "top",
          "right",
          "bottom",
          "left",
          "top",
          "right",
          "bottom",
        ]
      }
      isVisible={false}
      onBlur={[Function]}
      onDocumentClick={false}
      onDocumentKeyDown={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTriggerEnter={[Function]}
      placement="right"
      popper={
        <div
          className="pf-c-tooltip"
          id="pf-tooltip-1"
          role="tooltip"
          style={
            Object {
              "maxWidth": null,
              "opacity": 0,
              "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
            }
          }
        >
          <TooltipArrow />
          <TooltipContent
            isLeftAligned={false}
          >
            <span>
               You can manage other users' permissions with 'User access' 
            </span>
          </TooltipContent>
        </div>
      }
      popperMatchesTriggerWidth={false}
      positionModifiers={
        Object {
          "bottom": "pf-m-bottom",
          "left": "pf-m-left",
          "right": "pf-m-right",
          "top": "pf-m-top",
        }
      }
      trigger={
        <Label
          aria-describedby="pf-tooltip-1"
          color="purple"
        >
           Org. Administrator 
        </Label>
      }
      zIndex={9999}
    >
      <FindRefWrapper
        onFoundRef={[Function]}
      >
        <Label
          aria-describedby="pf-tooltip-1"
          color="purple"
        >
          <span
            aria-describedby="pf-tooltip-1"
            className="pf-c-label pf-m-purple"
          >
            <span
              className="pf-c-label__content"
            >
               Org. Administrator 
            </span>
          </span>
        </Label>
      </FindRefWrapper>
    </Popper>
  </Tooltip>
</StatusLabel>
`;
